csrc/host_ir/jit.cpp

```diff
@@ -529,37 +529,51 @@ void unpackInputs(
   }
 
   // Get the current function (main) and its input tensor array
-  llvm::Value* aten_tensor_array = func->getArg(1);
+  llvm::Value* main_func_input_array = func->getArg(1);
+  main_func_input_array->setName("input_args");
 
-  llvm::Type* aten_tensor_array_type = getInt8PtrDynamicArrayType(context);
+  llvm::Type* input_args_type = getInt8PtrDynamicArrayType(context);
   llvm::Type* tensor_type = getTensorPtrType(context);
 
   // bind input aten tensor sizes to val_to_value
   for (const auto [i, input] : enumerate(container->inputs())) {
-    auto* tv = dynamic_cast<TensorView*>(input);
-    NVF_ERROR(tv != nullptr, "Unsupported expression type: ", input);
-    llvm::Value* tensor_addr = builder.CreateGEP(
-        aten_tensor_array_type, aten_tensor_array, {builder.getInt64(i)});
-    tensor_addr->setName("input_aten_tensor_addr");
-    // Load the actual tensor pointer from the array
-    llvm::Value* tensor = builder.CreateLoad(tensor_type, tensor_addr);
-    tensor->setName("input_aten_tensor");
-    // bind input aten tensor sizes to val_to_value
-    // TODO: We should validate const size and strides here, ie. dim check
-    for (const auto [dim_idx, id] :
-         enumerate(TensorDomain::noReductions(tv->getLogicalDomain()))) {
-      if (id->isBroadcast()) {
-        val_to_value[id->extent()] = builder.getInt64(1);
-        if (id->hasExpandedExtent()) {
-          val_to_value[id->expandedExtent()] =
+    if (auto* tv = dynamic_cast<TensorView*>(input)) {
+      llvm::Value* tensor_addr = builder.CreateGEP(
+          input_args_type, main_func_input_array, {builder.getInt64(i)});
+      // Load the actual tensor pointer from the array
+      llvm::Value* tensor = builder.CreateLoad(tensor_type, tensor_addr);
+      tensor->setName("input_aten_tensor");
```

Comment:
As a future improvement, can we set the name to `ir_utils::varName(input)` so it's easier for human readers to map host IR to LLVM IR?